docs/components/truncate.md

```diff
@@ -0,0 +1,236 @@
+
+<KCard>
+  <template v-slot:body>
+    <KTruncate :rows="2">
+      <KBadge v-for="n in 25" :key="n">
+        Item {{ n }}
+      </KBadge>
+    </KTruncate>
+  </template>
+</KCard>
+
+```html
+<template>
+  <KTruncate :rows="2">
+    <KBadge v-for="n in 25" :key="n">
+      Item {{ n }}
+    </KBadge>
+  </KTruncate>
+</template>
+```
+
+### isTextContent
+
+By default the component treats anything passed through the `default` slot as collection of HTML elements. Use this prop if you want to truncate text.
+
+When this prop is set to `true`, the component applies different logic; truncation is achieved via the `-webkit-line-clamp` CSS property, rather than assessing the height of child elements to determine the row height.
+
+<KCard>
+  <template v-slot:body>
+    <KTruncate is-text-content :rows="3">
+      <p>A good design decision to apply text truncation would be when displaying a <b>large amount of textual content</b>, such as a list of articles or product descriptions, in a limited space, such as a mobile screen or a small widget. By truncating the text to a short summary, it is possible to present the information in a more organized and readable manner, allowing the user to quickly scan and understand the main points of each item.</p>
+      <p>The truncated text can also serve as a teaser, encouraging the user to click or tap to view the full content.</p>
+    </KTruncate>
+  </template>
+</KCard>
+
+```html
+<template>
+  <KTruncate is-text-content :rows="3">
+    <p>A good design decision to apply text truncation would be 
+      when displaying a <b>large amount of textual content</b>, such as a list 
+      of articles or product descriptions, in a limited space, such as 
+      a mobile screen or a small widget. By truncating the text to a 
+      short summary, it is possible to present the information in a more 
+      organized and readable manner, allowing the user to quickly scan 
+      and understand the main points of each item.</p>
+      <p>The truncated text can also serve as a teaser, encouraging the user to click or tap to view the full content.</p>
+  </KTruncate>
+</template>
+```
+
+### isExpanded
+
+If passed `true`, the component will be rendered expanded by default.
+
+<KCard>
+  <template v-slot:body>
+    <KTruncate is-expanded>
+      <KBadge v-for="n in 25" :key="n">
+        Item {{ n }}
+      </KBadge>
+    </KTruncate>
+  </template>
+</KCard>
+
+```html
+<template>
+  <KTruncate is-expanded>
+    <KBadge v-for="n in 25" :key="n">
+      Item {{ n }}
+    </KBadge>
+  </KTruncate>
+</template>
+```
+
+## Slots
+
+### default
+
+Slot for content.
+
+To ensure the best experience with the component, please pass elements of equal height in the default slot. The component will base its guess for the height of the element on the height of each individual sibling, so passing elements of equal height will result in a more accurate and consistent output.
```

Comment:
The explanation here is a little misleading if the user is providing paragraph text as the content. Does it make sense to note that here?

Reply:
Agree, maybe a little confusing -- added a note, specifying that this doesn't apply to text truncation scenarios